imap_processing/lo/l1b/lo_l1b.py

```diff
@@ -1004,3 +1045,247 @@ def create_datasets(
             )
 
     return dataset
+
+
+def initialize_l1b_histrates(
+    l1a_hist: xr.Dataset, attr_mgr_l1b: ImapCdfAttributes, logical_source: str
+) -> xr.Dataset:
+    """
+    Initialize the L1B histogram rates dataset.
+
+    Parameters
+    ----------
+    l1a_hist : xr.Dataset
+        The L1A histogram rates dataset.
+    attr_mgr_l1b : ImapCdfAttributes
+        Attribute manager used to get the L1B histogram rates dataset attributes.
+    logical_source : str
+        The logical source of the data product that's being created.
+
+    Returns
+    -------
+    l1b_histrates : xr.Dataset
+        The initialized L1B histogram rates dataset.
+    """
+    l1b_histrates = xr.Dataset(
+        attrs=attr_mgr_l1b.get_global_attributes(logical_source),
+    )
+
+    l1b_histrates["epoch"] = xr.DataArray(
+        l1a_hist["epoch"].values,
+        dims=["epoch"],
+        attrs=attr_mgr_l1b.get_variable_attributes("epoch"),
+    )
+    # Copy over fields from L1A DE that will not change in L1B processing
+    l1b_histrates["h_counts"] = xr.DataArray(
+        l1a_hist["hydrogen"].values,
+        dims=["epoch"],
+        # TODO: Add hydrogen to YAML file
+        # attrs=attr_mgr.get_variable_attributes("hydrogen"),
+    )
+    l1b_histrates["o_counts"] = xr.DataArray(
+        l1a_hist["oxygen"].values,
+        dims=["epoch"],
+        # TODO: Add oxygen to YAML file
+        # attrs=attr_mgr.get_variable_attributes("oxygen"),
+    )
```

Comment:
Here, if there are already attributes set in L1a I think you can do a direct insertion and get those transferred as well potentially.

```suggestion
    # Copy over fields from L1A DE that will not change in L1B processing
    l1b_histrates["h_counts"] = l1a_hist["hydrogen"]
    l1b_histrates["o_counts"] = l1a_hist["oxygen"]
```

Reply:
Agreed, but I'll create a ticket and do that in a separate PR. I need to change all of L1A histogram to use spin_bin instead of azimuth. To avoid having azimuth carry over into L1B right now, I'll make this change in that PR.

https://github.com/IMAP-Science-Operations-Center/imap_processing/issues/2462